From 9af92b9cd64cb4a2df8ad03685e38fedf4ca78b4 Mon Sep 17 00:00:00 2001
From: Ahmad Fatoum <a.fatoum@pengutronix.de>
Date: Wed, 21 Aug 2024 15:02:02 +0200
Subject: [PATCH 1/2] MIPS: dts: add /chosen/stdout-path where it is missing

A number of MIPS boards don't have a stdout-path set and instead rely on
CONFIG_CONSOLE_ACTIVATE_ALL/FIRST to enable a console.

This can get problematic if more consoles are added in future, or
CONFIG_CONSOLE_ACTIVATE_NONE is to be used. So on systems that currently
have a single console, let's have /chosen/stdout-path point at it.

Signed-off-by: Ahmad Fatoum <a.fatoum@pengutronix.de>
---
 arch/mips/dts/black-swift.dts   | 4 ++++
 arch/mips/dts/dlink-dir-320.dts | 4 ++++
 arch/mips/dts/img-ci20.dts      | 4 ++++
 arch/mips/dts/loongson-ls1b.dts | 4 ++++
 arch/mips/dts/rzx50.dts         | 4 ++++
 5 files changed, 20 insertions(+)

diff --git a/arch/mips/dts/black-swift.dts b/arch/mips/dts/black-swift.dts
index af817dbfc7e1..6bb26d427f2d 100644
--- a/arch/mips/dts/black-swift.dts
+++ b/arch/mips/dts/black-swift.dts
@@ -13,6 +13,10 @@ aliases {
 		spiflash = &spiflash;
 	};
 
+	chosen {
+		stdout-path = &uart;
+	};
+
 	memory@0 {
 		device_type = "memory";
 		reg = <0x00000000 0x4000000>;
diff --git a/arch/mips/dts/dlink-dir-320.dts b/arch/mips/dts/dlink-dir-320.dts
index d19c3087d017..beec40f3dcc5 100644
--- a/arch/mips/dts/dlink-dir-320.dts
+++ b/arch/mips/dts/dlink-dir-320.dts
@@ -7,6 +7,10 @@ / {
 	#address-cells = <1>;
 	#size-cells = <1>;
 
+	chosen {
+		stdout-path = &serial0;
+	};
+
 	memory@0 {
 		device_type = "memory";
 		reg = <0x00000000 0x2000000>;
diff --git a/arch/mips/dts/img-ci20.dts b/arch/mips/dts/img-ci20.dts
index e1d3280c4e11..71038890d977 100644
--- a/arch/mips/dts/img-ci20.dts
+++ b/arch/mips/dts/img-ci20.dts
@@ -14,6 +14,10 @@ / {
 	model = "MIPS Creator CI20";
 	compatible = "img,ci20";
 
+	chosen {
+		stdout-path = &uart0;
+	};
+
 	memory@0 {
 		device_type = "memory";
 		reg = <0x0 0x10000000>;
diff --git a/arch/mips/dts/loongson-ls1b.dts b/arch/mips/dts/loongson-ls1b.dts
index 89cce5636bcf..30520a34f53b 100644
--- a/arch/mips/dts/loongson-ls1b.dts
+++ b/arch/mips/dts/loongson-ls1b.dts
@@ -6,6 +6,10 @@ / {
 	model = "Loongson Tech LS1B Demo Board";
 	compatible = "loongson,ls1b";
 
+	chosen {
+		stdout-path = &serial2;
+	};
+
 	oscillator: oscillator {
 			#clock-cells = <0>;
 			compatible = "fixed-clock";
diff --git a/arch/mips/dts/rzx50.dts b/arch/mips/dts/rzx50.dts
index bd3631f7c451..93211a97a59f 100644
--- a/arch/mips/dts/rzx50.dts
+++ b/arch/mips/dts/rzx50.dts
@@ -6,6 +6,10 @@ / {
 	model = "Ritmix RZX-50";
 	compatible = "ritmix,rzx50";
 
+	chosen {
+		stdout-path = &serial1;
+	};
+
 	memory@0 {
 		device_type = "memory";
 		reg = <0x00000000 0x4000000>;
-- 
2.39.2

